api/src/server/auth.rs

```diff
@@ -208,81 +164,50 @@ pub async fn route_login(
     client_map: web::Data<ClientMap>,
     params: web::Path<(String,)>,
     query: web::Query<LoginQuery>,
-    sessions: web::Data<RwLock<Sessions>>,
     pool: web::Data<Pool>,
     identity: Identity,
 ) -> Result<HttpResponse, actix_web::Error> {
-    let provider_name = params.0.to_string();
-    let oidc_client = client_map.get_client(&provider_name)?;
-
+    let provider_name: &str = &params.0;
+    let oidc_client = client_map.get_client(provider_name)?;
     let conn = &pool.get().map_err(ResponseError::from)? as &PgConnection;
 
-    match request_token(oidc_client, query).await {
-        Ok(Some((token, userinfo))) => {
-            let sub = userinfo.sub.clone().unwrap();
+    // Send the user's code to the server to authenticate it
+    let (_, userinfo) = request_token(oidc_client, &query.code).await?;
 
-            let existing_user_provider =
-                UserProvider::filter_by_sub_and_provider(&sub, &provider_name)
-                    .get_result::<UserProvider>(conn)
-                    .optional()
-                    .map_err(ResponseError::from)?;
-            match existing_user_provider {
-                Some(user_provider) => match user_provider.user_id {
-                    Some(user_id) => {
-                        // User already exists so normal login
-                        let user: User = users::table
-                            .filter(users::dsl::id.eq(user_id))
-                            .get_result(conn)
-                            .map_err(ResponseError::from)?;
-                        login_user(
-                            &user_provider,
-                            user,
-                            token,
-                            userinfo,
-                            identity,
-                            sessions,
-                        )
-                    }
-                    None => {
-                        // no user account associated with this user_provider
-                        // yet so make one (they have logged in but did not set
-                        // username)
-                        init_user(
-                            &user_provider,
-                            userinfo,
-                            token,
-                            identity,
-                            sessions,
-                            conn,
-                        )
-                    }
-                },
-                None => {
-                    // user_provider not found (first login) so make the row
-                    // then init the user
-                    let user_provider: UserProvider = NewUserProvider {
-                        sub: &sub,
-                        provider_name: &provider_name,
-                    }
+    // Not sure when this can be None, hopefully never??
+    let sub: &str = userinfo.sub.as_ref().unwrap();
```

Comment:
It *should* never be None but i hope it would error before that in `request_token` if is not there